sql/src/main/java/io/crate/metadata/blob/InternalBlobTableInfoFactory.java

```diff
@@ -71,18 +75,25 @@ private IndexMetaData resolveIndexMetaData(String tableName, ClusterState state)
 
     @Override
     public BlobTableInfo create(TableIdent ident, ClusterService clusterService) {
-        IndexMetaData indexMetaData = resolveIndexMetaData(ident.name(), clusterService.state());
-        return new BlobTableInfo(
-            ident,
-            indexMetaData.getIndex(),
-            clusterService,
-            indexMetaData.getNumberOfShards(),
-            NumberOfReplicas.fromSettings(indexMetaData.getSettings()),
-            TableParameterInfo.tableParametersFromIndexMetaData(indexMetaData),
-            blobsPath(indexMetaData.getSettings()),
-            DocIndexMetaData.DEFAULT_ROUTING_HASH_FUNCTION,
-            Version.CURRENT,
-            null);
+        try {
+            IndexMetaData indexMetaData = resolveIndexMetaData(ident.name(), clusterService.state());
+            Map<String, Object> mappingMap = indexMetaData.mapping(Constants.DEFAULT_MAPPING_TYPE).getSourceAsMap();
+
+            return new BlobTableInfo(
+                ident,
+                indexMetaData.getIndex(),
+                clusterService,
+                indexMetaData.getNumberOfShards(),
+                NumberOfReplicas.fromSettings(indexMetaData.getSettings()),
+                TableParameterInfo.tableParametersFromIndexMetaData(indexMetaData),
+                blobsPath(indexMetaData.getSettings()),
+                DocIndexMetaData.getRoutingHashFunction(mappingMap),
+                DocIndexMetaData.getVersionCreated(mappingMap),
+                DocIndexMetaData.getVersionUpgraded(mappingMap));
+        } catch (Exception e) {
+            LOGGER.error("error extracting blob table info for {}", e, ident.fqn());
+            return null;
```

Comment:
is it save to return null here? also I'd only surround required stuff by try/catch to clarify the reason for that...